addons/mail/static/src/core/web/composer_patch.js

```diff
@@ -5,12 +5,152 @@ import { childNodes } from "@html_editor/utils/dom_traversal";
 import { Composer } from "@mail/core/common/composer";
 import { createDocumentFragmentFromContent } from "@mail/utils/common/html";
 
-import { markup } from "@odoo/owl";
+import { markup, toRaw, useEffect } from "@odoo/owl";
 
 import { patch } from "@web/core/utils/patch";
 import { renderToElement } from "@web/core/utils/render";
+import { fixInvalidHTML } from "@html_editor/utils/sanitize";
+import { MAIL_PLUGINS, MAIL_SMALL_UI_PLUGINS } from "./plugins/plugin_sets";
+import { Wysiwyg } from "@html_editor/wysiwyg";
+import { useService } from "@web/core/utils/hooks";
+import { isEmptyHtmlBody } from "@mail/utils/common/format";
+
+patch(Composer, {
+    components: { ...Composer.components, Wysiwyg },
+});
 
 patch(Composer.prototype, {
+    setup() {
+        super.setup();
+        this.wysiwyg = {
+            config: this.wysiwygConfigs,
+            editor: undefined,
+        };
+        this.suggestion = null;
+        useEffect(
+            (focus) => {
+                if (focus && this.wysiwyg.editor) {
+                    this.wysiwyg.editor.shared.selection.setCursorEnd(
+                        this.wysiwyg.editor.editable.lastChild
+                    );
+                    this.wysiwyg.editor.shared.selection.focusEditable();
+                    this.wysiwyg.editor.editable.focus();
+                }
+            },
+            () => [this.props.autofocus + this.props.composer.autofocus, this.props.placeholder]
+        );
+    },
+    get wysiwygConfigs() {
+        return {
+            content: fixInvalidHTML(this.props.composer.htmlBody) || "<p><br></p>",
+            placeholder: this.placeholder,
+            disableVideo: true,
+            Plugins: this.ui.isSmall ? MAIL_SMALL_UI_PLUGINS : MAIL_PLUGINS,
+            classList: ["o-mail-Composer-input", "o-mail-Composer-inputStyle"],
+            onChange: this.onChange.bind(this),
+            onBlur: this.onBlurWysiwyg.bind(this),
+            onEditorReady: () => {
+                if (this.isBodyEmpty) {
+                    this.restoreContent();
+                }
+                const body = fixInvalidHTML(this.props.composer.htmlBody);
+                if (!isEmptyHtmlBody(body)) {
+                    this.wysiwyg.editor.editable.innerHTML = body;
```

Comment:
Use safer `setElementContent` (make sure `fixInvalidHTML` is markup-aware internally if that's not already the case)